### PR TITLE
Remove never needed nor used JED Update Server from new installs

### PR DESCRIPTION
**Pull request by zero-24**
### Summary of Changes

Following Michaels tip: https://github.com/joomla/joomla-cms/pull/13041#issuecomment-263612114

> we should just delete that never used JED channel already) 

Remove never needed nor used JED Update Server from new installs

### Testing Instructions

- Install staging.
- go to the update sites view
- check theat the Joomla! Extension Directory entry is there
- install this branch: https://github.com/zero-24/joomla-cms/archive/remove_jedUpdateserv.zip
- check that this update server is not there.

### Documentation Changes Required

None as it was never used.

see: https://github.com/joomla/update.joomla.org/pull/31

### CODE REVIEW

**Comment by andrepereiradasilva** [comment]
shouldn't be deleted also on update?

```sql
DELETE FROM `#__updates`
WHERE `update_site_id` = (SELECT `update_site_id` FROM `#__update_sites` WHERE `name` = 'Joomla! Extension Directory' AND `type` = 'collection');

DELETE FROM `#__update_sites_extensions`
WHERE `update_site_id` = (SELECT `update_site_id` FROM `#__update_sites` WHERE `name` = 'Joomla! Extension Directory' AND `type` = 'collection');

DELETE FROM `#__update_sites`
WHERE `name` = 'Joomla! Extension Directory' AND `type` = 'collection';
```

**Comment by alikon** [comment]
is only for new install

**Comment by andrepereiradasilva** [comment]
also you need to remove them from:
- https://github.com/joomla/joomla-cms/blob/staging/administrator/manifests/files/joomla.xml#L51
- https://github.com/joomla/joomla-cms/blob/staging/tests/unit/stubs/database/jos_update_sites.csv
- https://github.com/joomla/joomla-cms/blob/staging/tests/unit/stubs/database/jos_update_sites_extensions.csv

**Comment by andrepereiradasilva** [comment]
> is only for new install

Why?

**Comment by alikon** [comment]
looking at https://update.joomla.org/jed/list.xml 
```xml
<extensionset name="JED" description="Joomla! Extension Directory">
<!--  TODO: fill this out  -->
</extensionset>
```

don't know why

**Comment by andrepereiradasilva** [comment]
@alikon what i mean is this update sites in all current joomla installs, so should be removed on updated too IMHO.

**Comment by zero-24** [comment]
Thanks. I can fix this later. I have also missed one place to remove that link ;) in the xml. The update part will be done via php as i have no idea how postgres or mssql works with sub querys ;)

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on fa48100dac480f6b5d15884952a9179ba47c8e78<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13047">issues.joomla.org/tracker/joomla-cms/13047</a>.</sub>

File: administrator/components/com_admin/script.php [hunk 1]
```diff
@@ -203,6 +204,38 @@ protected function uninstallEosPlugin()
 	}
 
 	/**
+	 * Remove the never used JED Updateserver
+	 *
+	 * @return  void
```
Comment (andrepereiradasilva) [inline administrator/components/com_admin/script.php]: misses `__DEPLOY_VERSION__`

File: administrator/components/com_admin/script.php [hunk 2]
```diff
@@ -203,6 +204,38 @@ protected function uninstallEosPlugin()
...
+				->delete($db->quoteName('#__update_sites'))
+				->where('update_site_id = ' . $id)
+		)->execute();
+
+		// Delete from update sites extensions
+		$result = $db->setQuery(
+			$db->getQuery(true)
+				->delete($db->quoteName('#__update_sites_extensions'))
+				->where('update_site_id = ' . $id)
+		)->execute();
```
Comment (andrepereiradasilva) [inline administrator/components/com_admin/script.php]: not essencial, but why only quoteName the database table and not the db fields?

File: installation/sql/sqlazure/joomla.sql [hunk 3]
```diff
@@ -2855,8 +2855,6 @@ SET IDENTITY_INSERT [#__update_sites] ON;
 INSERT INTO [#__update_sites] ([update_site_id], [name], [type], [location], [enabled], [last_check_timestamp])
 SELECT 1, 'Joomla! Core', 'collection', 'https://update.joomla.org/core/list.xml', 1, 0
 UNION ALL
-SELECT 2, 'Joomla! Extension Directory', 'collection', 'https://update.joomla.org/jed/list.xml', 1, 0
-UNION ALL
 SELECT 3, 'Accredited Joomla! Translations', 'collection', 'https://update.joomla.org/language/translationlist_3.xml', 1, 0
 UNION ALL
 SELECT 4, 'Joomla! Update Component Update Site', 'extension', 'https://update.joomla.org/core/extensions/com_joomlaupdate.xml', 1, 0;
```
Comment (andrepereiradasilva) [inline installation/sql/sqlazure/joomla.sql]: shoudn't the ids be changed here too?

**Comment by zero-24** [comment]
should be good now @andrepereiradasilva

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 830681f7c1c3bbfa82c386f1c99f786c93e2538a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13047">issues.joomla.org/tracker/joomla-cms/13047</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 830681f7c1c3bbfa82c386f1c99f786c93e2538a<br /><br />on code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13047">issues.joomla.org/tracker/joomla-cms/13047</a>.</sub>

**Comment by jeckodevelopment** [comment]
I have tested this item :white_check_mark: successfully on 830681f7c1c3bbfa82c386f1c99f786c93e2538a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13047">issues.joomla.org/tracker/joomla-cms/13047</a>.</sub>

**Comment by jeckodevelopment** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13047">issues.joomla.org/tracker/joomla-cms/13047</a>.</sub>

**Pull request merged**